### ISSUE TITLE
Initializing a map with zoom > 14 prevent it from rendering

### ISSUE THREAD

**Issue by farfromrefug** [CartoDB/mobile-sdk#276]
In my app i save map position /zoom to open back at the same position.
It triggered a bug in Carto Mobile SDK.
If i open my app, and immediately set the zoom to 15, then my carto layers wont render.
For a while i thought it was me, but i now understand why the 15. 
I think it's related to ```DEFAULT_MAX_ZOOM``` and carto online datasources ```maxZoom``` beeing set to it. If the map starts with a zoom past ```maxZoom```, then the datasource/layer do not render. Wont even render on map move. You need to zoom out ```< maxZoom``` for the map to start rendering.

I did not test it yet with carto mobile sample apps. As usual i am working with a [wrapper](https://github.com/farfromrefug/nativescript-carto) around Carto Mobile SDK in [Nativescript](https://www.nativescript.org/)

**Comment by mtehver** [comment]
This is quite unlikely to be a bug in the SDK. SDK does not ask query data sources for tiles with zoom > tileDataSource.max_zoom_level. Could come up with a sample demonstrating this?

**Comment by farfromrefug** [comment]
@mtehver thanks for answering this.
Found out more about that bug. It is actually an issue with the new ```OrderedTileDataSource```
Managed to reproduce it in the Android Kotlin sample app.
i modified the ```PackageDownloadActivity```  by adding this in ```onResume```
```kotlin
val dataSource = com.carto.datasources.OrderedTileDataSource(
                com.carto.datasources.PackageManagerTileDataSource(contentView?.manager),
                com.carto.datasources.CartoOnlineTileDataSource("carto.streets")
        );
        val layer =  com.carto.layers.VectorTileLayer(dataSource, com.carto.layers.CartoOnlineVectorTileLayer(CartoBaseMapStyle.CARTO_BASEMAP_STYLE_VOYAGER).tileDecoder)
        layer.isPreloading = true
        this.mapView.layers.add(layer)
        val washingtonDC = this.mapView.options.baseProjection.fromWgs84(MapPos(-77.0369, 38.9072))

        // Set specific focus, zoom, tilt and rotation for an optimal initial view
        this.mapView.setFocusPos(washingtonDC, 0.0f)
        this.mapView.setZoom(17.0f,0.0f)
```
Make sure you disable the default ```onlineLayer``` or you will see this layer.

Now when opening the ```Offline Map``` activity the map wont render until you zoom out under 14.

I will try to setup your iOS dev env and see if i can fix this

**Comment by mtehver** [comment]
This should be fixed in 4.2.0RC1, `OrderedTileDataSource` did not properly forward 'replace with parent flag'. Closing it.
